persist/fs/retriever_test.go

```diff
@@ -92,7 +93,26 @@ type streamResult struct {
 	stream     xio.SegmentReader
 }
 
+// TestBlockRetrieverHighConcurrentSeeks tests the retriever with high
+// concurrent seeks, but without caching the shard indices. This means that the
+// seekers will be opened lazily by calls to ConcurrentIDBloomFilter() in the
+// SeekerManager
 func TestBlockRetrieverHighConcurrentSeeks(t *testing.T) {
+	testBlockRetrieverHighConcurrentSeeks(t, false)
+}
+
+// TestBlockRetrieverHighConcurrentSeeksCacheShardIndices tests the retriever
+// with high concurrent seekers and calls cache shard indices at the beginning.
+// This means that the seekers will be opened all at once in the beginning and
+// by the time ConcurrentIDBloomFilter() is called, they seekers will already be
+// open.
+func TestBlockRetrieverHighConcurrentSeeksCacheShardIndices(t *testing.T) {
+	testBlockRetrieverHighConcurrentSeeks(t, true)
+}
+
+func testBlockRetrieverHighConcurrentSeeks(t *testing.T, shouldCacheShardIndices bool) {
+	defer leaktest.CheckTimeout(t, 2*time.Minute)()
```

Comment:
Hm 2 minutes seems like a long time, any reason not to just use the default?

Reply:
it times out in travis haha. I think the primary culprit is the race detector making it slow.

Reply:
the checkTimeout start at the beginning of the test unfortunately, not when the test is complete.

Reply:
I could probably move it to a separate file and add the "big" tag or whatever, but doesn't seem worth it right now

Reply:
Ah I see, yeah - time from start of the test is pretty unfortunate.